File: backend/src/main/java/com/corebanking/modules/customer/service/CustomerService.java
package com.corebanking.modules.customer.service;

import com.corebanking.modules.customer.dto.CustomerFilterParams;
import com.corebanking.modules.customer.dto.CustomerRequest;
import com.corebanking.modules.customer.dto.CustomerResponse;
import com.corebanking.modules.customer.entity.CustomerStatus;
import com.corebanking.shared.PagedResponse;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface CustomerService {

    PagedResponse<CustomerResponse> findAll(CustomerFilterParams filters, Pageable pageable);

    CustomerResponse findById(UUID id);

    CustomerResponse create(CustomerRequest request);

    CustomerResponse update(UUID id, CustomerRequest request);

    void delete(UUID id);

    CustomerResponse updateStatus(UUID id, CustomerStatus status);
}

File: frontend/src/services/customer.service.ts
import { apiClient } from "@/lib/axios";
import type { PagedResponse } from "@/types";
import type { Customer, CustomerFilterParams, CustomerRequest } from "@/types/customer.types";

const BASE = "/v1/customers";

export const customerService = {
  findAll: async (params: CustomerFilterParams): Promise<PagedResponse<Customer>> => {
    const { data } = await apiClient.get<PagedResponse<Customer>>(BASE, { params });
    return data;
  },

  findById: async (id: string): Promise<Customer> => {
    const { data } = await apiClient.get<Customer>(`${BASE}/${id}`);
    return data;
  },

  create: async (payload: CustomerRequest): Promise<Customer> => {
    const { data } = await apiClient.post<Customer>(BASE, payload);
    return data;
  },

  update: async (id: string, payload: CustomerRequest): Promise<Customer> => {
    const { data } = await apiClient.put<Customer>(`${BASE}/${id}`, payload);
    return data;
  },

  delete: async (id: string): Promise<void> => {
    await apiClient.delete(`${BASE}/${id}`);
  },

  updateStatus: async (id: string, status: "ACTIVO" | "INACTIVO"): Promise<Customer> => {
    const { data } = await apiClient.patch<Customer>(`${BASE}/${id}/status`, null, {
      params: { status },
    });
    return data;
  },
};
